src/test/suite/fixtures/recorded/languages/typescript/takeEveryItem5.yml

```diff
@@ -0,0 +1,45 @@
+spokenForm: take every item
+languageId: typescript
+command:
+  actionName: setSelection
+  partialTargets:
+    - type: primitive
+      modifier: {type: containingScope, scopeType: collectionItem, includeSiblings: true}
+  extraArgs: []
+marks: {}
+initialState:
+  documentContents: |-
+    const value = {
+        key1: "hello",
+        key2: "there",
+    };
+  selections:
+    - anchor: {line: 1, character: 10}
+      active: {line: 1, character: 10}
+    - anchor: {line: 2, character: 10}
+      active: {line: 2, character: 17}
+finalState:
+  documentContents: |-
+    const value = {
+        key1: "hello",
+        key2: "there",
+    };
+  selections:
+    - anchor: {line: 1, character: 4}
+      active: {line: 1, character: 17}
+    - anchor: {line: 2, character: 4}
+      active: {line: 2, character: 17}
+    - anchor: {line: 1, character: 4}
+      active: {line: 1, character: 17}
+    - anchor: {line: 2, character: 4}
+      active: {line: 2, character: 17}
```

Comment:
Is this really what we want? Appears selections are duplicated

Reply:
Agreed,  but has actually nothing to do with this pr. transformSelection/getNodeMatcher gets called for every selection. But might be a good idea to actually filter out duplicate selections at the end of the pipeline. You should probably put that on the task list for the new grammar.